3696/1/name_to_link_transformer.rb

```diff
@@ -0,0 +1,27 @@
+class NameToLinkTransformer
+  attr_reader :valid, :link
+  def initialize(name)
+    @name = name
+    @valid = false
+    @link = ''
+    github_link
+  end
+
+  def github_link
+    doc = Nokogiri::HTML(URI.open("https://rubygems.org/gems/#{@name}"))
+    github_in_code_link(doc)
+    github_in_home_link(doc) unless @valid
+  end
+
+  def github_in_code_link(page)
+    all_links = page.css('a#code').map { |link| link.attribute('href').to_s }
+    @valid = !all_links.empty?
```

Comment:
make a separate validation method or return false here and process this case on the higher level